src/main/java/edu/harvard/iq/dataverse/MetadataBlock.java

```diff
@@ -101,14 +101,30 @@ public void setDatasetFieldTypes(List<DatasetFieldType> datasetFieldTypes) {
     }
     
     public boolean isDisplayOnCreate() {
+        // relying on "should" doesn't seem to work in context of a template 
```

Comment:
should relies on a transient datsetfieldtype localdisplayoncreate that must not be getting getting set in the DatasetVersionUI init method. If it were, it think it should work. Is it just because of the order of these 2 lines: https://github.com/IQSS/dataverse/blob/039b3b7266ad76b56b550d0eb3316c78f84d7125/src/main/java/edu/harvard/iq/dataverse/DatasetPage.java#L2135-L2136 - the updateDatasetFieldInputLevels sets localdisplayOncreate, right after the dvUI is inited. (Same ordering of these two calls in the DatasetPage.edit method.)

Reply:
I did try switching the order here and commenting out the section in init-UI but it still doesn't display the metadata block that has only display on create set at the DV level.

Reply:
Got it - I still wonder a bit about the order of those lines, but digging some more I realized why shouldDisplayOnCreate can't work in the MetadataBlock.isDisplayOnCreate call - the fieldTypes being checked there are straight from the db and not the ones from the workingVersion datasetfield.getDatasetFieldType list which has been localized to the local collection. Given that - it's clear why your solution is needed - the metadatablock localDisplayOnCreate has to be set by checking the localized field types from the working version, which is what the code in DatasetVersionUI does. 
In the small PR I made against this one, I tried to edit the comments and did a little cleanup to avoid double looping. - I made comments in the PR about it.